src/contexts/product-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useReducer } from "react";
+import { productsReducer } from "../components/reducers/product-reducer";
+import { products } from "../backend/db/products";
+import { getProductsInPriceRange, getProductsWithAppliedRating, getProductsInCategory, getSortedProducts } from "../utilities/sorting-functions";
+const ProductContext = createContext();
+const ProductProvider = ({ children }) => {
+    const [state, dispatch] = useReducer(productsReducer, {
+        priceRange: 1575,
+        lowToHigh: false,
+        highToLow: false,
+        rating: "",
+        categoryState: {
+            organicFruits: false, organicVegetables: false, organicSpices: false, organicGroceries: false, organicDairy: false, organicOils: false
+        },
+        productsInCategory: []
+    })
+    const productsInPriceRange = getProductsInPriceRange([...products], state);
+    const productsWithRating = getProductsWithAppliedRating(productsInPriceRange, state);
+    const categorizedProducts = getProductsInCategory(productsWithRating, state);
+    const filteredProducts = getSortedProducts(categorizedProducts, state);
```

Comment:
Instead of calling multiple functions together you can create a single function that takes all these filter utilities together and returns us the combined result of the providedProducts and state.
To make sense of what I am talking about you can [see this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#function_composition_enabling_piping)